UnityProject/Assets/Scripts/Construction/Computer.cs

```diff
@@ -43,21 +45,37 @@ public void ServerPerformInteraction(HandApply interaction)
 			$"{interaction.Performer.ExpensiveName()} disconnects the monitor.",
 			() =>
 			{
-				//drop all our contents
-				ItemStorage itemStorage = null;
-				// rare cases were gameObject is destroyed for some reason and then the method is called
-				if (gameObject != null)
-				{
-					itemStorage = GetComponent<ItemStorage>();
-				}
-
-				if (itemStorage != null)
-				{
-					itemStorage.ServerDropAll();
-				}
-				var frame = Spawn.ServerPrefab(framePrefab, SpawnDestination.At(gameObject)).GameObject;
-				frame.GetComponent<ComputerFrame>().ServerInitFromComputer(this);
-				Despawn.ServerSingle(gameObject);
+				WhenDestroyed(null);
 			});
 	}
+
+	private void Awake()
+	{
+		if (!CustomNetworkManager.IsServer) return;
+
+		integrity = GetComponent<Integrity>();
+
+		integrity.OnWillDestroyServer.AddListener(WhenDestroyed);
+	}
+
+	public void WhenDestroyed(DestructionInfo info)
+	{
+		//drop all our contents
+		ItemStorage itemStorage = null;
+		// rare cases were gameObject is destroyed for some reason and then the method is called
+		if (gameObject != null)
+		{
+			itemStorage = GetComponent<ItemStorage>();
+		}
+
+		if (itemStorage != null)
+		{
+			itemStorage.ServerDropAll();
+		}
+		var frame = Spawn.ServerPrefab(framePrefab, SpawnDestination.At(gameObject)).GameObject;
```

Comment:
We can enter this part of code with `gameObject == null`
I know that you just moved old code, but I think it might be a good time to fix this